code-challenges.test.js

```diff
@@ -14,45 +14,116 @@
-
-const filterLetterA = "a"
+describe("filteredWordsByLetters", () => {
+  it("returns an array of all the words containing that particular letter", () => {
+    const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+    const filterLetterA = "a"
 // Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
+    const filterLetterE = "e"
 // Expected output: ["Cherry", "Blueberry", "Peach"]
+  expect(filteredWordsByLetters(fruitArray, filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+  expect(filteredWordsByLetters(fruitArray, filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+  })
+})
+
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// use .filter() and includes() methods to filter
+
+const filteredWordsByLetters = (array, letter) => {
+  return array.filter(word => word.toLowerCase().includes(letter.toLowerCase()))
+}
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a "full house". A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
-const hand4 = [7, 2, 7, 2, 7]
-// Expected output: true
+describe("isAFullHouse", () => {
+  it("determines whether or not the array is a full house", () => {
+    const hand1 = [5, 5, 5, 3, 3]
+    // Expected output: true
+    const hand2 = [5, 5, 3, 3, 4]
+    // Expected output: false
+    const hand3 = [5, 5, 5, 5, 4]
+    // Expected output: false
+    const hand4 = [7, 2, 7, 2, 7]
+    // Expected output: true
+  expect(isAFullHouse(hand1)).toBe(true)  
+  expect(isAFullHouse(hand2)).toBe(false)  
+  expect(isAFullHouse(hand3)).toBe(false)  
+  expect(isAFullHouse(hand4)).toBe(true)  
+  })
+})
+
 
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// take in the array set a new varible called fullCounts = new Map()
+//create a for loop set up the iteration for each element being a num in the hand's array one by one
+// create an if, fullCounts.has(num) to check fullCounts 
+//  new Map() creates a new Map object
+//create an else if the num is not in fullCounts 
+// use .has() to return true if key exists in Map
+// use .set() to set the value for a key in the Map
+// use .get() to get the value for the Key in the Map
+// use Array.from to create an array from the values map.
+
+const isAFullHouse = (hand) => {
+  const fullCounts = new Map();
+
+  for (const num of hand) {
+    if (fullCounts.has(num)) {
+      fullCounts.set(num, fullCounts.get(num) + 1);
+    } else {
+      fullCounts.set(num, 1);
+    }
+  }
+
+  const counts = Array.from(fullCounts.values());
+
+  return counts.includes(2) && counts.includes(3);
+}
+
+
```

Comment:
I know you found exploring Map() interesting, and I like you enthusiasm! However, please use the concepts we use in class.  I want to make sure you understand the fundamentals.  This problem can be done with what we learned in class.  🙂. Great job explaining what each method does.